Intersect.Server/Entities/Entity.cs

```diff
@@ -3037,6 +3053,10 @@ public StatusPacket[] StatusPackets()
             return statusPackets;
         }
 
+        public virtual void OnNearEntityDies(Entity en) { }
+
+        public virtual void OnNearEntityAttacked(Entity victim, Entity attacker)  { }
```

Comment:
Rename to `OnSwarmAttacked`. Parameters can be left in for mergeability for those who need the extra information.